pkg/auth/oauth/external/github/github.go

```diff
@@ -98,3 +128,65 @@ func (p provider) GetUserIdentity(data *osincli.AccessData) (authapi.UserIdentit
 
 	return identity, true, nil
 }
+
+// getUserOrgs retrieves the organization membership for the user with the given access token.
+func getUserOrgs(token string) (sets.String, error) {
+	// start with the empty set, and the initial org url
+	userOrgs := sets.NewString()
+	orgURL := githubUserOrgURL
+
+	for {
+		// fetch organizations
+		organizations := []githubOrg{}
+		links, err := getJSON(orgURL, token, &organizations)
+		if err != nil {
+			return nil, err
+		}
+		for _, org := range organizations {
+			if len(org.Login) > 0 {
+				userOrgs.Insert(strings.ToLower(org.Login))
+			}
+		}
+
+		// see if we need to page
+		// https://developer.github.com/v3/#link-header
+		nextURL := links["next"]
+		if len(nextURL) == 0 {
+			// no next URL, we're done paging
+			break
+		}
+		if nextURL == orgURL {
+			// identical next URL, break to avoid a loop
```

Comment:
Based on the comment, I'd expect a `sets.String` to look for cycles.  Is this defensive?  If so, `sets.String` if you're going to keep it.  If its actual behavior, indicate that its the "common" path.